app/resources.py

```diff
@@ -148,76 +221,151 @@ class ItemResource(Resource):
-                item = Item.query.filter_by(bucket_id=id,
+                item = Item.query.filter_by(bucket_id=bucketlist.id,
                                             id=item_id).first()
                 if item:
                     return marshal(item, item_fields)
                 else:
                     return {'message': 'item does not exist'}, 404
             else:
-                items = Item.query.filter_by(bucket_id=id).all()
-                return marshal(items, item_fields)
+                if (not request.args.get('page')
+                        and not request.args.get('limit')):
+                    items = Item.query.filter_by(
+                        bucket_id=bucketlist.id
+                    ).all()
+                    return marshal(items, item_fields)
+                else:
+                    parser = reqparse.RequestParser()
+                    parser.add_argument('page',
+                                        type=int,
+                                        default=1,
+                                        location='args')
+                    parser.add_argument('limit',
+                                        type=int,
+                                        default=10,
+                                        location='args')
+                    parser.add_argument('q',
+                                        type=str,
+                                        location='args')
+                    args = parser.parse_args(strict=True)
+
+                    items = Item.query.filter(
+                        Item.bucket_id == bucketlist.id
+                    ).order_by(
+                        Item.id.asc()).paginate(
+                        args['page'], args['limit'], error_out=False)
+
+                    return {
+                        "count": len(items.items),
+                        "next": "/api/v1/bucketlists/{}"
+                        "/items?page={}&limit={}"
+                        .format(id, args['page'] + 1, args['limit']),
+                        "previous": "/api/v1/bucketlists/{}"
+                        "/items?page={}&limit={}"
+                        .format(id, args['page'] - 1, args['limit']),
+                        "items": marshal(items.items,
+                                         item_fields)}
+
         else:
             return {'message': 'bucketlist does not exist'}, 404
 
     @jwt_required()
     def post(self, id):
         parser = reqparse.RequestParser()
-        parser.add_argument('title', type=str, required=True, location='json')
-        parser.add_argument('description', type=str, required=True, location='json')
+        parser.add_argument('title', type=str,
+                            required=True, location='json')
+        parser.add_argument('description', type=str,
+                            required=True, location='json')
 
         args = parser.parse_args(strict=True)
         if len(args['title'].strip()) == 0 or len(
```

Comment:
` args = parser.parse_args(strict=True)` raises an AttributeError. This is because I  specified 'title' as a required parameter in 
`parser.add_argument('title', type=str, required=True, location='json')
`

. The error is handled by flask-restful
by sending a response containing a message with the text, Missing required parameter in the JSON body.